src/util/triplea/mapXmlCreator/MapPropertiesPanel.java

```diff
@@ -0,0 +1,278 @@
+package util.triplea.mapXmlCreator;
+
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.io.File;
+
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+import games.strategy.common.swing.SwingAction;
+import util.image.FileOpen;
+
+
+public class MapPropertiesPanel {
+
+  public static void layout(final MapXmlCreator mapXMLCreator) {
+
+    if (!MapXmlHelper.getXmlStringsMap().containsKey("info_@name")) {
+      for (final String newKey : new String[] {"info_@name", "info_@version"}) {
+        MapXmlHelper.putXmlStrings(newKey, "");
+      }
+    }
+
+    if (MapXmlHelper.getResourceList().isEmpty())
+      MapXmlHelper.addResourceList("");
+
+    final JTextField textFieldMapName = new JTextField(MapXmlHelper.getXmlStringsMap().get("info_@name"));
+    final JTextField textFieldMapVersion = new JTextField(MapXmlHelper.getXmlStringsMap().get("info_@version"));
+    final JTextField textFieldResourceName = new JTextField(MapXmlHelper.getResourceList().get(0));
+    final JTextField textFieldMapImageFile =
+        new JTextField((MapXmlCreator.mapImageFile == null ? "" : MapXmlCreator.mapImageFile.getAbsolutePath()));
+    final JTextField textFieldCentersFile = new JTextField(
+        (MapXmlCreator.mapCentersFile == null ? "" : MapXmlCreator.mapCentersFile.getAbsolutePath()));
+    final JTextField textFieldWaterFilter = new JTextField(MapXmlCreator.waterFilterString);
+    final JLabel labelMapName = new JLabel("Map Name:");
+    final JLabel labelMapNameExample = new JLabel("e.g. 'Revised'");
+    final JLabel labelMapVersion = new JLabel("Map Version:");
+    final JLabel labelMapVersionExample = new JLabel("e.g. '1.2.0.1'");
+    final JLabel labelResourceName = new JLabel("Resource Name:");
+    final JLabel labelResourceNameExample = new JLabel("e.g. 'PUs'");
+    final JLabel labelMapImageFile = new JLabel("Map Image File:");
+    final JButton buttonSelectMapImageFile = new JButton("Browse");
+    final JLabel labelCentersFile = new JLabel("Map Centers File:");
+    final JButton buttonSelectCentersFile = new JButton("Browse");
+    final JLabel labelWaterFilter = new JLabel("Water Territory Filter");
+    final JLabel labelWaterFilterExample = new JLabel("e.g. 'Sea Zone'");
+
+    GridBagLayout gbl_panel_1 = new GridBagLayout();
+    gbl_panel_1.rowHeights = new int[] {32, 32, 32, 32, 32, 32, 32, 0};
+    gbl_panel_1.rowWeights = new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
```

Comment:
Magic numbers, construct these values so that:
A) it is clear why we have the number of "0.0" as we do
B) it is linked to whatever is driving this value, so if we updated something somewhere else, we don't have to magically know that this also needs to be updated.